pkg/cmd/server/etcd/etcd.go

```diff
@@ -135,7 +135,7 @@ func TestNewEtcdClient(etcdClient newetcdclient.Client) error {
 // NeverRetryOnFailure is a retry function for the etcdClient.  If there's only one machine, master election doesn't make much sense,
 // so we don't bother to retry, we simply dump the failure and return the error directly.
 func NeverRetryOnFailure(cluster *etcdclient.Cluster, numReqs int, lastResp http.Response, err error) error {
-	if len(cluster.Machines) > 1 {
+	if len(cluster.Machines) > 1 || numReqs < 2 {
```

Comment:
> I'm testing whether the assumption that a local etcd server doesn't go through leader election is false.

I don't think that's what this is doing.  If you look here: https://github.com/openshift/origin/blob/master/Godeps/_workspace/src/github.com/coreos/etcd/client/client.go#L294-L296, they simply assign that error to any unknown internal error blindly.  Attempting to re-run the request isn't giving you that information.